python/lsst/ap/association/diaPipe.py

```diff
@@ -1167,14 +1167,22 @@ def loadRefreshedDiaObjects(self, region, preloadedDiaObjects):
         refreshedDiaObjects = convertDataFrameToSdmSchema(self.schema, diaObjects, tableName="DiaObject")
 
         refreshedIsInPreloaded = refreshedDiaObjects.index.isin(preloadedDiaObjects.index)
+        preloadedIsInRefreshed = preloadedDiaObjects.index.isin(refreshedDiaObjects.index)
         nUniqueRefreshed = (~refreshedIsInPreloaded).sum()
+        nUniquePreloaded = (~preloadedIsInRefreshed).sum()
         if nUniqueRefreshed > 0:
             self.log.info("Reloading the diaObject table during association yielded "
                           "an additional %d objects over the %d preloaded diaObjects.",
                           nUniqueRefreshed, len(preloadedDiaObjects))
+        if nUniquePreloaded == 0:
             return refreshedDiaObjects
-        else:
+        elif nUniqueRefreshed == 0:
             return preloadedDiaObjects
+        else:
+            # We can get "preloaded" diaObjects that don't appear in the APDB with
+            # the CI datasets in ap_verify, since those are stored as a dataset and
+            # an empty APDB is created. In that case, combine the two catalogs.
+            return pd.concat([refreshedDiaObjects, preloadedDiaObjects.loc[~preloadedIsInRefreshed]])
```

Comment:
This approach seems a bit dangerous IRL, since this case would trigger if objects were deleted from the APDB. The "backwards" logic of the preceding two cases also seems like a maintainability risk.

Would it be simpler to just turn off the refresh in `ap_verify`?

Reply:
@isullivan pointed out on Slack that the pre-preloaded catalogs are also intended to simulate history _external to_ the data set, so we can't remove them without losing functionality.

It looks like the risk of unexpected behavior in production might be the least of the available evils.